### DISCUSSION TITLE
Is there a way to handle multiple themes and subthemes with Unistyles?

### DISCUSSION THREAD

**Discussion by ryanalencar** [jpudysz/react-native-unistyles#355] [Q&A]
I'm currently working on a React Native design system library with multiple tokens such as **colors** and **measures**.

And this project is going to scale a lot, so in the future we could have n themes with **light** and **dark** subthemes.

I was wondering what is the best way to manage this using Unistyles.

Currently I'm doing in a linear way and is not being easy to handle theme and subtheme changing.

```
const themes= {
  defaultLight: { colors: { ...defaultLight }, measures },
  defaultDark: { colors: { ...defaultDark }, measures },
  agroamigoLight: { colors: { ...agroamigoLight }, measures },
  agroamigoDark: { colors: { ...agroamigoDark }, measures },
};

UnistylesRegistry.addThemes(bnbDsTheme);
```

I tried creating a structure like this, but without success.

```
const themeObjects = {
  default: {
    light: defaultLight,
    dark: defaultDark,
  },
  agroamigo: {
    light: agroamigoLight,
    dark: agroamigoDark,
  },
};
```

**Comment by jpudysz** [comment]
The only way to register themes is with `UnistylesRegistry.addThemes`. If you have dozens of them, there is no other helper to ease things up. Keep in mind that this is custom case, most users use 1-2 themes per app.
